salt/cloud/clouds/digital_ocean.py

```diff
@@ -663,6 +663,27 @@ def show_keypair(kwargs=None, call=None):
     return details
 
 
+def import_keypair(kwargs=None, call=None):
+    '''
+    Upload public key to cloud provider.
+    Similar to EC2 import_keypair.
+
+    kwargs
+        file(mandatory): public key file-name
+        keyname(mandatory): public key name in the provider
+    '''
+    with open(kwargs['file'], 'r') as public_key_filename:
```

Comment:
Can you please change the `open` call to use Salt's `fopen` function instead? See [here](https://github.com/saltstack/salt/blob/b6d640da0db7a5a57709b48d8dc1d19509901e32/salt/utils/__init__.py#L1182) for reference. The prevents leaking file descriptors.